endpoint/platform/wp77xx/wp77xx.c

```diff
@@ -0,0 +1,236 @@
+  tty.c_lflag &= ~(ECHO | ECHONL | ICANON | ISIG | IEXTEN);
+  tty.c_oflag &= ~OPOST;
+
+  /* fetch bytes as they become available */
+  tty.c_cc[VMIN] = 1;
+  tty.c_cc[VTIME] = 1;
+
+  if (tcsetattr(fd, TCSANOW, &tty) != 0) {
+    LE_ERROR("Error from tcsetattr: %s\n", strerror(errno));
+    return SC_ENDPOINT_UART_SET_ATTR;
+  }
+  return SC_OK;
+}
+
+static status_t uart_init(const char *device) {
+  if (device == NULL) device = DEFAULT_PORT;
+  int fd;
+
+  fd = open(device, O_RDWR | O_NOCTTY | O_SYNC);
+  if (fd < 0) {
+    LE_ERROR("Error opening %s: %s\n", device, strerror(errno));
+    return SC_ENDPOINT_UART;
+  }
+  /*baudrate 115200, 8 bits, no parity, 1 stop bit */
+  set_interface_attribs(fd, B115200);
+  uart_fd = fd;
+
+  return SC_OK;
+}
+
+static void uart_write(const int fd, const char *cmd) {
+  /* simple output */
+  ssize_t cmd_len = strlen(cmd);
+  ssize_t wlen = write(fd, cmd, cmd_len);
+  if (wlen != cmd_len) {
+    LE_ERROR("Error from write: %zd, %s\n", wlen, strerror(errno));
+  }
+  tcdrain(fd); /* delay for output */
+}
+
+static char *uart_read(const int fd) {
+  unsigned char buf[READ_BUFFER_SIZE];
+  char *response = NULL;
+
+  ssize_t rdlen = read(fd, buf, sizeof(buf) - 1);
+  if (rdlen > 0) {
+    LE_DEBUG("uart read: %s", buf);
+    response = (char *)malloc(sizeof(char) * rdlen);
+    strncpy(response, (char *)buf, READ_BUFFER_SIZE);
+  } else if (rdlen < 0) {
+    LE_ERROR("Error from read: %zd: %s\n", rdlen, strerror(errno));
+  }
+
+  return response;
+}
+
+static void uart_clean(const int fd) {
+  if (tcflush(fd, TCIOFLUSH) != 0) {
+    LE_ERROR("tcflush error");
+  }
+}
+
+status_t sec_init(void) {
+  LE_INFO("Initialize secure storage success");
+  return SC_OK;
+}
+
+status_t sec_write(const char *name, const uint8_t *buf, size_t buf_size) {
+  le_result_t result = le_secStore_Write(name, buf, buf_size);
+
+  switch (result) {
+    case LE_UNAVAILABLE:
+      return SC_ENDPOINT_SEC_UNAVAILABLE;
+    case LE_OK:
+      return SC_OK;
+    default:
+      return SC_ENDPOINT_SEC_FAULT;
+  }
+}
+
+status_t sec_read(const char *name, uint8_t *buf, size_t *buf_size) {
+  le_result_t result = le_secStore_Read(name, buf, buf_size);
+
+  switch (result) {
+    case LE_NOT_FOUND:
+      return SC_ENDPOINT_SEC_ITEM_NOT_FOUND;
+    case LE_UNAVAILABLE:
+      return SC_ENDPOINT_SEC_UNAVAILABLE;
+    case LE_OK:
+      return SC_OK;
+    default:
+      return SC_ENDPOINT_SEC_FAULT;
+  }
+}
+
+status_t sec_delete(const char *name) {
+  le_result_t result = le_secStore_Delete(name);
+
+  switch (result) {
+    case LE_NOT_FOUND:
+      return SC_ENDPOINT_SEC_ITEM_NOT_FOUND;
+    case LE_UNAVAILABLE:
+      return SC_ENDPOINT_SEC_UNAVAILABLE;
+    case LE_OK:
+      return SC_OK;
+    default:
+      return SC_ENDPOINT_SEC_FAULT;
+  }
+}
+
+static const struct device_operations wp77xx_ops = {
+    .init = wp77xx_init, .fini = wp77xx_release, .get_key = wp77xx_get_key, .get_device_id = wp77xx_get_device_id};
```

Comment:
Append `,` (comma) right after `wp77xx_get_device_id` and split each item into new lines.